Add a way to use cprofile to generate stats and render as flamegraph

kimishpatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #1225

Summary:
While trying to debug slow export I found using cprofile quite useful.
However, it is much more intuitive to use icicle kind of flamegraphs.
This PR adds this util by building on top of snakeviz

Test Plan:
Tested locally

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D51380270](https://our.internmc.facebook.com/intern/diff/D51380270)

kimishpatel committed:
Add a way to use cprofile to generate stats and render as flamegraph

Summary:
While trying to debug slow export I found using cprofile quite useful.
However, it is much more intuitive to use icicle kind of flamegraphs.
This PR adds this util by building on top of snakeviz

Test Plan:
Tested locally

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51380270).

kimishpatel committed:
Update on "Add a way to use cprofile to generate stats and render as flamegraph"


Summary:
While trying to debug slow export I found using cprofile quite useful.
However, it is much more intuitive to use icicle kind of flamegraphs.
This PR adds this util by building on top of snakeviz

Test Plan:
Tested locally

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D51380270](https://our.internmc.facebook.com/intern/diff/D51380270)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51380270).

kimishpatel committed:
Update on "Add a way to use cprofile to generate stats and render as flamegraph"


Summary:
While trying to debug slow export I found using cprofile quite useful.
However, it is much more intuitive to use icicle kind of flamegraphs.
This PR adds this util by building on top of snakeviz

Test Plan:
Tested locally

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D51380270](https://our.internmc.facebook.com/intern/diff/D51380270)

[ghstack-poisoned]

kimishpatel committed:
Update on "Add a way to use cprofile to generate stats and render as flamegraph"


Summary:
While trying to debug slow export I found using cprofile quite useful.
However, it is much more intuitive to use icicle kind of flamegraphs.
This PR adds this util by building on top of snakeviz

Test Plan:
Tested locally

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D51380270](https://our.internmc.facebook.com/intern/diff/D51380270)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51380270).

mcr229 commented on util/cprofiler_flamegraph.md:
```diff
@@ -0,0 +1,17 @@
+
+## Installation
+
+pip install snakeviz
```
Create an install_requirements.sh for this?

kimishpatel replied:
It is just snakeviz pip install, so didnt quite do it.

kimishpatel committed:
Update on "Add a way to use cprofile to generate stats and render as flamegraph"


Summary:
While trying to debug slow export I found using cprofile quite useful.
However, it is much more intuitive to use icicle kind of flamegraphs.
This PR adds this util by building on top of snakeviz

Test Plan:
Tested locally

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D51380270](https://our.internmc.facebook.com/intern/diff/D51380270)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51380270).

kimishpatel committed:
Update on "Add a way to use cprofile to generate stats and render as flamegraph"


Summary:
While trying to debug slow export I found using cprofile quite useful.
However, it is much more intuitive to use icicle kind of flamegraphs.
This PR adds this util by building on top of snakeviz

Test Plan:
Tested locally

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D51380270](https://our.internmc.facebook.com/intern/diff/D51380270)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51380270).

kimishpatel committed:
Update on "Add a way to use cprofile to generate stats and render as flamegraph"


Summary:
While trying to debug slow export I found using cprofile quite useful.
However, it is much more intuitive to use icicle kind of flamegraphs.
This PR adds this util by building on top of snakeviz

Test Plan:
Tested locally

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D51380270](https://our.internmc.facebook.com/intern/diff/D51380270)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51380270).

kimishpatel committed:
Update on "Add a way to use cprofile to generate stats and render as flamegraph"


Summary:
While trying to debug slow export I found using cprofile quite useful.
However, it is much more intuitive to use icicle kind of flamegraphs.
This PR adds this util by building on top of snakeviz

Test Plan:
Tested locally

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D51380270](https://our.internmc.facebook.com/intern/diff/D51380270)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51380270).

kimishpatel committed:
Update on "Add a way to use cprofile to generate stats and render as flamegraph"


Summary:
While trying to debug slow export I found using cprofile quite useful.
However, it is much more intuitive to use icicle kind of flamegraphs.
This PR adds this util by building on top of snakeviz

Test Plan:
Tested locally

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D51380270](https://our.internmc.facebook.com/intern/diff/D51380270)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51380270).
